lib/queryable_ordered_set.dart

```diff
@@ -28,10 +28,19 @@ class _CacheEntry<C, T> {
 /// find a handful of elements, specially if this is done every tick, you
 /// can use this class, that pays a small O(number of registers) cost on [add],
 /// but lets you find (specific) subsets at O(0).
+///
+/// Note that you can change [strictMode] to allow for querying for unregistered
+/// types; if you do so, the registration cost is payed on the first query.
 class QueryableOrderedSet<T> extends OrderedSet<T> {
+  /// Controls whether running an unregistered query throws an error or
+  /// performs a just-in-time filtering.
+  final bool strictMode;
   final Map<Type, _CacheEntry<T, T>> _cache = {};
 
-  QueryableOrderedSet([int Function(T e1, T e2)? compare]) : super(compare);
+  QueryableOrderedSet({
```

Comment:
sadly dart doesn't support optional and named parameters together for some reason :(